initramfs-suspend

```diff
@@ -0,0 +1,61 @@
+#!/bin/sh
+# Copyright 2013 Vianney le Clément de Saint-Marcq <vleclement@gmail.com>  
+# Copyright 2017 Zhongfu Li <me@zhongfu.li>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with This program.  If not, see http://www.gnu.org/licenses/.
+
+cryptname="${1}"
+
+# Start udev from initramfs
+/lib/systemd/systemd-udevd --daemon --resolve-names=never
+
+# Sync one more time, just in case
+sync
+
+# Suspend root device
+[ -z "${cryptname}" ] || cryptsetup luksSuspend "${cryptname}"
+
+# Suspend the system
+echo mem > /sys/power/state
+
+# Resume root device
+TMP_FILE=/run/new_key
+. /etc/ykluks.cfg
+
+[ -z "${cryptname}" ] ||
+    while true; do
+	P1=$(/lib/cryptsetup/askpass "$WELCOME_TEXT")
+
+	if [ "$HASH" = "1" ]; then
+		P1=$(printf %s "$P1" | sha256sum | awk '{print $1}')
+	fi
+
+	R="$(ykchalresp -2 "$P1" 2>/dev/null || true)"
+
+	touch "$TMP_FILE"
+	chmod 600 "$TMP_FILE"
+
+	if [ "$CONCATENATE" = "1" ]; then
+		echo -n "$P1$R" > "$TMP_FILE"
+	else
+		echo -n "$R" > "$TMP_FILE"
+	fi
+
+        cryptsetup --key-file="$TMP_FILE" luksResume "${cryptname}"
+	rm -rf "$TMP_FILE"
```

Comment:
Should we shred the file like here?
https://github.com/cornelinux/yubikey-luks/blob/master/yubikey-luks-enroll#L86

Reply:
The problem is AFAIK shred binary isn't available in initramfs by default so if we want to use it we have to add it explicitly in https://github.com/cornelinux/yubikey-luks/blob/master/hook#L21 . In my opinion this is unnecessary complication. I don't have expertise about how likely is to recover file created in initramfs and deleted by simple rm. If I or someone else find info about this or provide better solution it can be adjusted. Until then I would left it as it is.

Reply:
I figured that we can actually get rid of tmp file completely, see https://github.com/Vincent43/yubikey-luks/commit/c3a828a0cfd5753b443e77ef03bf73a30f37b2ea

We can adjust all of the rest occurrences of it the same way.